samza-core/src/main/java/org/apache/samza/storage/StorageManagerUtil.java

```diff
@@ -139,13 +161,26 @@ public static boolean isOffsetFileValid(File storeDir, Set<SystemStreamPartition
    * @param storeName the store name to use
    * @param taskName the task name which is referencing the store
    * @param offsets The SSP-offset to write
+   * @param isSideInput true if store is a side-input store, false if it is a regular store
    * @throws IOException because of deserializing to json
    */
   public static void writeOffsetFile(File storeBaseDir, String storeName, TaskName taskName, TaskMode taskMode,
-      Map<SystemStreamPartition, String> offsets) throws IOException {
-    File offsetFile = new File(getStorePartitionDir(storeBaseDir, storeName, taskName, taskMode), OFFSET_FILE_NAME);
+      Map<SystemStreamPartition, String> offsets, boolean isSideInput) throws IOException {
+
+    // First, we write the new-format offset file
+    File offsetFile = new File(getStorePartitionDir(storeBaseDir, storeName, taskName, taskMode), OFFSET_FILE_NAME_NEW);
     String fileContents = OBJECT_WRITER.writeValueAsString(offsets);
     FileUtil.writeWithChecksum(offsetFile, fileContents);
+
+    // Now we write the old format offset file, which are different for store-offset and side-inputs
+    if (isSideInput) {
+      offsetFile = new File(getStorePartitionDir(storeBaseDir, storeName, taskName, taskMode), SIDE_INPUT_OFFSET_FILE_NAME_LEGACY);
+      fileContents = OBJECT_WRITER.writeValueAsString(offsets);
+      FileUtil.writeWithChecksum(offsetFile, fileContents);
+    } else {
+      offsetFile = new File(getStorePartitionDir(storeBaseDir, storeName, taskName, taskMode), OFFSET_FILE_NAME_LEGACY);
+      FileUtil.writeWithChecksum(offsetFile, offsets.entrySet().iterator().next().getValue());
```

Comment:
Would it be good to have some validation here to make sure there is exactly one entry in `offsets`?
Or maybe have two different methods: one for side inputs which takes in an `offsets` map and one for non-side-inputs which takes in a single `offset`.

Reply:
yup, 
i'll start a separate PR with these changes, and tests